src/test/java/executor/service/controllers/ScenarioSourceControllerTest.java

```diff
@@ -0,0 +1,195 @@
+package executor.service.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import executor.service.model.Scenario;
+import executor.service.model.Step;
+import executor.service.services.ScenarioSourceListenerImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+@WebMvcTest(ScenarioSourceController.class)
+class ScenarioSourceControllerTest {
+
+    private ObjectMapper objectMapper;
+    private Scenario testScenario;
+    private List<Scenario> testScenarioList;
+
+    @MockBean
+    private ScenarioSourceListenerImpl ScenarioSourceListenerImpl;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @BeforeEach
+    public void setUp() throws IOException {
+        Step step1 = new Step("clickCSS", "body > ul > li:nth-child(1) > a");
+        Step step2 = new Step("clickXpath", "/html/body/p");
+        Step step3 = new Step("sleep", "5000");
+        Step step4 = new Step("clickCSS", "body > ul > li:nth-child(1) > a");
+        Step step5 = new Step("clickXpath", "/html/body/p");
+        Scenario tempScenario = new Scenario("test scenario 2", "http://info.cern.ch", Arrays.asList(step1, step2, step3));
+        testScenario = new Scenario("test scenario 1", "http://info.cern.ch", Arrays.asList(step4, step5));
+        testScenarioList = Arrays.asList(testScenario, tempScenario);
+        objectMapper = new ObjectMapper();
+    }
+
+    @Test
+    void testAddValidScenarioStatusIsOK() throws Exception {
```

Comment:
let's use common code style
```suggestion
    public void testAddValidScenarioStatusIsOK() throws Exception {
```

Reply:
@artemkedyk done